src/lib.rs

```diff
@@ -1186,6 +1194,26 @@ impl Block {
             .collect::<Vec<Fragment>>()
             .into()
     }
+
+    pub fn epoch(&self) -> u32 {
+        self.0.date().epoch
+    }
+
+    pub fn slot(&self) -> u32 {
+        self.0.date().slot_id
+    }
+
+    pub fn chain_length(&self) -> u32 {
+        u32::from(self.0.chain_length())
+    }
+
+    pub fn leader_id(&self) -> Option<PoolId> {
```

Comment:
it may be useful to document that this can be null if the node is a genesis block or if it is a bft block